lib/index.d.ts

```diff
@@ -0,0 +1,286 @@
+// Type definitions for stardog.js 0.3.1
+// Project: https://github.com/stardog-union/stardog.js
+// Definitions by: Stephen Nowell <http://github.com/snowell>, Ty Soehngen <http://github.com/tysoeh>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.1
+
+import Headers from 'fetch-ponyfill';
+import Guid from '@types/node';
+
+declare global {
+  
+  /** stardog.js: The Stardog JS API*/
+  namespace Stardog {
+
+    type ContentType =
+        'application/x-turtle' |
+        'text/turtle' |
+        'application/rdf+xml' |
+        'text/plain' |
+        'application/x-trig' |
+        'text/x-nquads' |
+        'application/trix';
+    
+    type MIMEType =
+        'text/plain' |
+        'application/json' |
+        'text/boolean';
+
+    interface HTTPOptions {
+        headers?: Headers;
+        method?: string;
+        mimeType?: MIMEType;
+        contentType?: ContentType;
+    }
+
+    interface QueryOptions {
+        query: string;
+        baseURI?: string;
+        limit?: number;
+        offset?: number;
+        reasoning?: boolean;
+    }
+
+    interface HTTPMessage {
```

Comment:
You can leave this for now, but let's move to just always use HTTPBody and set it to `null` for things that don't really have bodies. The typing on that is going to be object OR boolean OR string OR null which isn't great but is what it is.